src/vm/dllimportcallback.cpp

```diff
@@ -1412,6 +1412,8 @@ VOID UMThunkMarshInfo::RunTimeInit()
         new (&sigInfo) PInvokeStaticSigInfo(pMD);
     else
         new (&sigInfo) PInvokeStaticSigInfo(GetSignature(), GetModule());
+    if (pStubMD == NULL)
+        m_cbActualArgSize += numRegistersUsed * STACK_ELEM_SIZE;
     if (sigInfo.GetCallConv() == pmCallConvCdecl)
```

Comment:
Yes, either fix UMThunkStub; or change what m_cbActualArgSize means on Linux x86 (update the comment next to it, and check that there are no other places that depend on the current meaning.